uzdevums-2/LoanManagementApi/LoanManagementApi/Repositories/StatisticsRepository.cs

```diff
@@ -0,0 +1,88 @@
+﻿using LoanManagementApi.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace LoanManagementApi.Repositories
+{
+    public class StatisticsRepository : IStatisticsRepository
+    {
+        private readonly ManagementDBContext _ctx;
+
+        public StatisticsRepository(ManagementDBContext ctx)
+        {
+            _ctx = ctx;
+        }
+
+        public async Task<Balance> GetUserBalancceStatus(int id)
+   {
+            var query = await
+               (from u in _ctx.Users
+                where u.UserId == id
+                select new { u.GivenLoans, u.TakenLoans })
+               .FirstOrDefaultAsync();
+
+            if (query == null)
+            {
+                return null;
+            }
+
+            var gLoans = query.GivenLoans != null ? query.GivenLoans.ToList().Sum(g => g.Amount) : 0;
+            var tLoans = query.TakenLoans != null ? query.TakenLoans.ToList().Sum(t => t.Amount) : 0;
+
+            var result = new Balance
+            {
+                Status = gLoans == tLoans ? BalanceStatus.Zero : gLoans > tLoans ? BalanceStatus.Positive : BalanceStatus.Negative
+            };
+
+            return result;
+        }
+
+        public async Task<UserInfo> GetBiggestBorrower()
+        
+        {
+
+            var query = from user in _ctx.Users
```

Comment:
šeit ļoti gribējās redzēt linq method syntax, nevis query syntax - vai vismaz vienu no tiem, nevis abus. Protams, tas ir subjektīvs jautājums, bet miksēt, atkal subjektīvi, tomēr nevajadzētu. Vismaz mēs to savos projektos cenšamies ievērot.